app/stac_api/validators.py

```diff
@@ -18,63 +20,108 @@
         'image/tiff; application=geotiff; profile=cloud-optimized',
         'Cloud Optimized GeoTIFF (COG)', ['.tiff', '.tif']
     ),
-    ('application/pdf', 'PDF', ['.pdf']),
-    ('application/x.pdf+zip', 'Zipped PDF', ['.zip']),
-    ('application/json', 'JSON', ['.json']),
-    ('application/x.json+zip', 'Zipped JSON', ['.zip']),
-    ('application/x-netcdf', 'NetCDF', ['.nc']),
-    ('application/x.netcdf+zip', 'Zipped NetCDF', ['.zip']),
-    ('application/xml', 'XML', ['.xml']),
-    ('application/x.xml+zip', 'Zipped XML', ['.zip']),
-    ('application/vnd.mapbox-vector-tile', 'mbtiles', ['.mbtiles']),
-    ('text/plain', 'Text', ['.txt']),
-    ('text/x.plain+zip', 'Zipped text', ['.zip']),
-    ('application/x.dwg+zip', 'Zipped DWG', ['.zip']),
-    ('application/zip', 'Generic Zip File', ['.zip']),
-    ('image/tiff', 'TIFF', ['.tiff', '.tif']),
-    ('image/jpeg', 'JPEG', ['.jpeg', '.jpg']),
-    ('image/png', 'PNG', ['.png']),
-    ('application/vnd.sqlite3', 'sqlite', ['.sqlite']),
+    MediaType('application/pdf', 'PDF', ['.pdf']),
+    MediaType('application/x.pdf+zip', 'Zipped PDF', ['.zip']),
+    MediaType('application/json', 'JSON', ['.json']),
+    MediaType('application/x.json+zip', 'Zipped JSON', ['.zip']),
+    MediaType('application/x-netcdf', 'NetCDF', ['.nc']),
+    MediaType('application/x.netcdf+zip', 'Zipped NetCDF', ['.zip']),
+    MediaType('application/xml', 'XML', ['.xml']),
+    MediaType('application/x.xml+zip', 'Zipped XML', ['.zip']),
+    MediaType('application/vnd.mapbox-vector-tile', 'mbtiles', ['.mbtiles']),
+    MediaType('text/plain', 'Text', ['.txt']),
+    MediaType('text/x.plain+zip', 'Zipped text', ['.zip']),
+    MediaType('application/x.dwg+zip', 'Zipped DWG', ['.zip']),
+    MediaType('application/zip', 'Generic Zip File', ['.zip']),
+    MediaType('image/tiff', 'TIFF', ['.tiff', '.tif']),
+    MediaType('image/jpeg', 'JPEG', ['.jpeg', '.jpg']),
+    MediaType('image/png', 'PNG', ['.png']),
+    MediaType('application/vnd.sqlite3', 'sqlite', ['.sqlite']),
 ]
-MEDIA_TYPES_MIMES = [x[0] for x in MEDIA_TYPES]
-MEDIA_TYPES_EXTENSIONS = [ext for media_type in MEDIA_TYPES for ext in media_type[2]]
-MEDIA_TYPES_BY_TYPE = {media[0]: media for media in MEDIA_TYPES}
+
+quoted_string_pattern = re.compile('^"[^"]+"$')
+
+
+def get_mime_comparable(mime_string):
+    '''
+    Generates a tuple out of a MIME string that will be equal to all equivalent MIME strings.
+
+    The rules for equivalence are:
+    - Types, subtypes and parameter names are case insensitive.
+    - Parameter values may be case sensitive (this function interprets this statement as "always
+      case sensitive")
+    - Order of the parameters is not important
+    - no whitespace around "=" allowed. Whitespace only allowed around the ";" parameter separator
+    - Parameter values are allowed to be quoted
+    '''
+    full_type, *arguments = map(lambda x: x.strip(), mime_string.split(';'))
+    full_type = full_type.lower()
```

Comment:
The `lower()` could/should be integrated in the lambda `lambda x: x.strip().lower()`
`full_type` is misleading, it contains only the type without parameters, should be `media_type`. 
I don't really like the mix of naming of *media type* and *mime type* or *mime string*. MIME type is the former name for media type and is the same thing. I would keep the new name media type everywhere (even though mime is a shorter name).
So instead of `mime_string` could be `media_type_str` or simply `media_type`

Reply:
If I put lower() in the lambda, it will also lowercase the the parameter values, and according to the specs, if a parameter value is case independent depends on the parameter name, and that's too complicated to implement, that's why I decided to keep all parameter values case dependent.

Reply:
The goal was to distinguish the MediaType tuple object and the media type string. So I renamed the "mime_string" to "media_type_str"